1-Authentication/1-sign-in/README.md

```diff
@@ -0,0 +1,365 @@
+
+### Dynamic scopes and incremental consent
+
+In **Azure AD**, the scopes (permissions) set directly on the application registration are called static scopes. Other scopes that are only defined within the code are called dynamic scopes. This has implications on the **login** (i.e. loginPopup, loginRedirect) and **acquireToken** (i.e. `acquireTokenPopup`, `acquireTokenRedirect`, `acquireTokenSilent`) methods of **MSAL.js**. Consider:
+
+```javascript
+     const loginRequest = {
+          scopes: [ "openid", "profile", "User.Read" ]
+     };
+
+     const tokenRequest = {
+          scopes: [ "Mail.Read" ]
+     };
+
+     // will return an ID Token and an Access Token with scopes: "openid", "profile" and "User.Read"
+     instance.loginPopup(loginRequest);
+
+     // will fail and fallback to an interactive method prompting a consent screen
+     // after consent, the received token will be issued for "openid", "profile" ,"User.Read" and "Mail.Read" combined
+     instance.acquireTokenPopup(tokenRequest);
+```
+
+In the code snippet above, the user will be prompted for consent once they authenticate and receive an **ID Token** and an **Access Token** with scope `User.Read`. Later, if they request an **Access Token** for `User.Read`, they will not be asked for consent again (in other words, they can acquire a token *silently*). On the other hand, the user did not consented to `Mail.Read` at the authentication stage. As such, they will be asked for consent when requesting an **Access Token** for that scope. The token received will contain all the previously consented scopes, hence the term *incremental consent*.
+
+### Acquire a Token
+
+**MSAL.js** exposes 3 APIs for acquiring a token: `acquireTokenPopup()`, `acquireTokenRedirect()` and `acquireTokenSilent()`. The `acquireTokenSilent()` API is meant to retrieve a non-expired access token from cache *silently*.
+
+```javascript
+export function App() {
+    const { instance, accounts, inProgress } = useMsal();
+    const account = useAccount(accounts[0]);
+    const [apiData, setApiData] = useState(null);
+
+    useEffect(() => {
+        if (account) {
+            instance.acquireTokenSilent({
+                scopes: ["User.Read"],
+                account: account
+            }).then((response) => {
+                if(response) {
+                    callMsGraph(response.accessToken).then((result) => setApiData(result));
```

Comment:
>callMsGraph [](start = 20, length = 11)

this methods details warrant another section, like sending AT in header , the API URL and how we process the response

Reply:
makes sense